### ISSUE TITLE
remote debugging not always showing "loaded modules"

### ISSUE THREAD

**Issue by flat-eric147** [microsoft/ClearScript#338]
Hello!

I am running a V8 engine inside my C# application and I use the V8ScriptEngineFlags.EnableDebugging | V8ScriptEngineFlags.EnableRemoteDebugging flags on script engine creation. I start VS Code and connect to the remote debugging port (9222 for instance) and "sometimes" it works and I get to see my code in the "Loaded Modules" section in VS.Code. I can step into the code etc, it works as expected! However, when I disconnect and reconnect or even restart my application AND VS.Code it would connect but it won't show me anything under "loaded modules". The whole thing resumes to work again after a couple of minutes, as if something is caching/blocking somewhere something (?). Don't really know where to look into to debug this issue. 
this is all running on a local machine under win10 with all the latest libs/updates.
Anybody else experiencing this? Any idea where I can start to troubleshoot?

**Comment by ClearScriptLib** [comment]
Hi @flat-eric147,

Thanks for reporting this. We'll take a look. In the meantime, do you know whether this issue affects Chrome/Edge DevTools? Or is it specific to VSCode?

Thanks!

**Comment by flat-eric147** [comment]
Thanks for you reply. To be honest I never tried with DevTools, I just tried now and it _seems_ slightly more stable but it still happens that the debugger sees "old" code (from the previous V8 engine instance). I suspected the .net garbge collector perhaps not cleaning up the v8 engine properly, so I tried to add explicitly GC.Collect() at the end of the cleanup sequence, something like:

`
engine.Interrupt();
engine.CollectGarbage(true);
engine.Dispose();
engine = null;

GC.Collect();
`

But still no glory.
